include/sqlpp11/detail/remove_cvref.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023, Roland Bock
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ *   Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ *   Redistributions in binary form must reproduce the above copyright notice, this
+ *   list of conditions and the following disclaimer in the documentation and/or
+ *   other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SQLPP11_DETAIL_REMOVE_CVREF_H
+#define SQLPP11_DETAIL_REMOVE_CVREF_H
+
+#include <type_traits>
+
+namespace sqlpp
+{
+  namespace detail
+  {
+    template<typename T>
+    using remove_reference_t = typename std::remove_reference<T>::type;
```

Comment:
Do you have plans to use this? Otherwise, this whole header seems a bit overkill. `remove_cvref_t` is used in one location, everything else is not used at all?

Reply:
remove_cvref uses the other two, that's why i also added those. I assumed you used headers like this to emulate c++14+ features like enable_if_t. I'll inline it at the only use.

Reply:
Ah, that's not what I meant. Sorry for being unclear:

I am all for `using` declarations and creating headers, but only if it makes a real difference. In this it is adding a lot of overhead just to be able to say `remove_cvref_t<T>` instead of `typename std::remove_cv<typename std::remove_reference<T>::type>::type` in a single place.

Note: There is also `std::decay` which should do the job, too?

Reply:
`decay` is a superset of `remove_cvref` and does things we don't need here (array, function pointer). It would work exactly the same though.

I added a comment with suggestions of the different variations, feel free to pick one of those :)

Reply:
Cool, I was not aware of this suggestion feature. That's nice.

And thanks for playing through the options! Much appreciated!